app/presenters/documents_info_export_presenter.rb

```diff
@@ -0,0 +1,86 @@
+class DocumentsInfoExportPresenter
+  attr_reader :lead_organisation, :type
+
+  def initialize(lead_organisation:, type:)
+    @lead_organisation = lead_organisation
+    @type = type
+  end
+
+  def call
+    document_ids.sort.map do |id|
+      { document_id: id,
+        document_information: {
+          locales: locales(id),
+          subtypes: subtypes(id),
+          lead_organisations: lead_orgs(id)
+        } }
+    end
+  end
+
+  private
+
+  def document_ids
+    @document_ids ||= type.constantize
+      .joins(:organisations)
+      .where(edition_organisations: { organisation_id: organisation_id, lead: true })
+      .where(type: type)
+      .latest_edition
+      .pluck(:document_id)
+  end
+
+  def organisation_id
+    Organisation.find_by(content_id: lead_organisation).id
+  end
+
+  def locales(doc_id)
+    locales_hash[doc_id].split(',')
+  end
+
+  def subtypes(doc_id)
+    if type_has_subtype?
+      subtypes_hash[doc_id].split(',').map do |subtype_id|
+        (type + 'Type').constantize.find_by_id(subtype_id.to_i).singular_name
+      end
+    end
+  end
+
+  def lead_orgs(doc_id)
+    lead_orgs_hash[doc_id].split(',')
+  end
+
+  def locales_hash
+    @locales_hash ||= Edition
+      .joins(:translations)
+      .where(document_id: document_ids)
+      .group(:document_id)
+      .pluck(:document_id, "GROUP_CONCAT(DISTINCT(edition_translations.locale))")
+      .to_h
```

Comment:
rather than doing `.to_h` it might be better to do `.each_with_object({}) { |(k, v), memo| memo[k] = v.split(",") }` as that will mean you don't have to worry about the array split later.

Reply:
Yup sure